src/components/application_manager/test/commands/mobile/on_button_event_notification_test.cc

```diff
@@ -0,0 +1,157 @@
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "gtest/gtest.h"
+#include "utils/shared_ptr.h"
+#include "smart_objects/smart_object.h"
+#include "application_manager/smart_object_keys.h"
+
+#include "application_manager/mock_application_manager.h"
+#include "application_manager/mock_application.h"
+
+#include "command_impl.h"
+#include "commands/commands_test.h"
+#include "mobile/on_button_event_notification.h"
+
+#include "interfaces/MOBILE_API.h"
+#include "application_manager/application.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace mobile_commands_test {
+
+namespace am = ::application_manager;
+namespace commands = am::commands;
+namespace mobile_result = mobile_apis::Result;
+
+using ::testing::_;
+using ::testing::Return;
+using namespace mobile_apis;
+using am::commands::MessageSharedPtr;
+using am::commands::mobile::OnButtonEventNotification;
+
+typedef ::utils::SharedPtr<OnButtonEventNotification> NotificationPtr;
+
+namespace {
+const int app_id1 = 10;
+const int app_id2 = 8;
+const bool full_screen = true;
+const bool app_id_valid = true;
+const bool send_message_expected = true;
+}
+
+class OnButtonEventNotificationTest
+    : public CommandsTest<CommandsTestMocks::kIsNice> {
+ protected:
+  OnButtonEventNotificationTest()
+      : command_(CreateCommand<OnButtonEventNotification>()) {}
+
+  void CheckExpectations(const bool is_app_id_valid,
+                         const HMILevel::eType app_hmi_level,
+                         const size_t app_id1,
+                         const size_t app_id2,
+                         const bool is_full_screen = false,
+                         const bool is_send_message_expected = false) {
+    MockAppPtr mock_app(CreateMockApp());
+    MockAppPtr subscribed(CreateMockApp());
+    std::vector<ApplicationSharedPtr> subscribedApps;
+    subscribedApps.push_back(subscribed);
+
+    if (is_app_id_valid) {
+      EXPECT_CALL(app_mngr_, application(_)).WillOnce(Return(mock_app));
+    } else {
+      EXPECT_CALL(app_mngr_, application(_))
+          .WillOnce(Return(ApplicationSharedPtr()));
+    }
+    EXPECT_CALL(app_mngr_, applications_by_button(_))
+        .WillOnce(Return(subscribedApps));
+    EXPECT_CALL(*subscribed, hmi_level()).WillRepeatedly(Return(app_hmi_level));
+    EXPECT_CALL(*subscribed, app_id()).WillRepeatedly(Return(app_id1));
+    EXPECT_CALL(*mock_app, app_id()).WillRepeatedly(Return(app_id2));
+    if (is_full_screen) {
+      EXPECT_CALL(*subscribed, IsFullscreen()).WillOnce(Return(true));
+    }
+
+    EXPECT_CALL(app_mngr_, SendMessageToMobile(_, _))
+        .Times(is_send_message_expected);
```

Comment:
@OHerasym TImes(bool)? Are you sure that is it is correct? At least it is not clear